xnmt/inferences.py

```diff
@@ -150,6 +150,7 @@ def _generate_output(self, generator: 'models.GeneratorModel', src_file: str,
     with open(trg_file, 'wt', encoding='utf-8') as fp:  # Saving the translated output to a trg file
       cur_sent_i = 0
       src_batch, ref_batch, assert_batch = [], [], []
+      event_trigger.set_reporting(self.reporter is not None)
```

Comment:
How about also calling ``event_trigger.set_reporting(False)`` after inference is finished? That way, we don't have to worry about e.g. having a reporter in a dev checkpoint that would then potentially cause training to slow down.

Reply:
Nice idea. Implemented.